File: POS_10Apr2025/POS/App_Code/clsDepartment.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Globalization
Imports System.Collections.Generic
Imports System.Threading
Public Class Controller_clsDepartment
    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
    Public Sub New()
    End Sub
#Region "Property"

    Private _Department_id As Integer
    Private _Name As String
    Private _Value As Integer
    Private _Is_Active As String
    Private _Cmp_id As Integer
    Private _Ip_Address As String
    Private _Login_id As Integer
    Private _Department_category_id As Integer
    Private oclsDepartment As clsDepartment

    Public Property Login_id() As Integer
        Get
            Return _Login_id
        End Get
        Set(ByVal value As Integer)
            _Login_id = value
        End Set
    End Property
    Public Property Department_category_id() As Integer
        Get
            Return _Department_category_id
        End Get
        Set(ByVal value As Integer)
            _Department_category_id = value
        End Set
    End Property
    Public Property Ip_Address() As String
        Get
            Return _Ip_Address
        End Get
        Set(ByVal value As String)
            _Ip_Address = value
        End Set
    End Property

    Public Property Cmp_id() As Integer
        Get
            Return _Cmp_id
        End Get
        Set(ByVal value As Integer)
            _Cmp_id = value
        End Set
    End Property

    Public Property Is_Active() As String
        Get
            Return _Is_Active
        End Get
        Set(ByVal value As String)
            _Is_Active = value
        End Set
    End Property

    Public Property Value() As Integer
        Get
            Return _Value
        End Get
        Set(ByVal value As Integer)
            _Value = value
        End Set
    End Property

    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property

    Public Property Department_id() As Integer
        Get
            Return _Department_id
        End Get
        Set(ByVal value As Integer)
            _Department_id = value
        End Set
    End Property
#End Region
#Region "Fuction"
    Public Function Insert() As Boolean
        Try
            oclsDepartment = New clsDepartment()
            oclsDepartment.Department_id = Department_id
            oclsDepartment.Name = Name
            oclsDepartment.Value = Value
            oclsDepartment.Cmp_id = Cmp_id
            oclsDepartment.Login_id = Login_id
            oclsDepartment.Ip_Address = Ip_Address
            oclsDepartment.Department_category_id = Department_category_id
            If oclsDepartment.Insert() Then
                Return True
            End If
            Return False

        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try

    End Function
    Public Function Update() As Boolean
        Try
            oclsDepartment = New clsDepartment()
            oclsDepartment.Department_id = Department_id
            oclsDepartment.Name = Name
            oclsDepartment.Value = Value
            oclsDepartment.Cmp_id = Cmp_id
            oclsDepartment.Login_id = Login_id
            oclsDepartment.Ip_Address = Ip_Address
            oclsDepartment.Department_category_id = Department_category_id
            If oclsDepartment.Update() Then
                Return True
            End If
            Return False

        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try

    End Function
    Public Function Delete() As Boolean
        Try
            oclsDepartment = New clsDepartment()
            oclsDepartment.Department_id = Department_id
            oclsDepartment.Name = Name
            oclsDepartment.Value = Value
            oclsDepartment.Cmp_id = Cmp_id
            oclsDepartment.Login_id = Login_id
            oclsDepartment.Ip_Address = Ip_Address
            oclsDepartment.Department_category_id = Department_category_id
            If oclsDepartment.Delete() Then
                Return True
            End If
            Return False

        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try

    End Function
    Public Function [Select]() As DataTable
        Dim dt As DataTable
        Try
            oclsDepartment = New clsDepartment()
            oclsDepartment.Department_id = Department_id
            oclsDepartment.Cmp_id = Cmp_id
            dt = oclsDepartment.[Select]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [QrList]() As DataTable
        Dim dt As DataTable
        Try
            oclsDepartment = New clsDepartment()

            oclsDepartment.Cmp_id = Cmp_id
            dt = oclsDepartment.[QrList]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [Select_By_Cmp]() As DataTable
        Dim dt As DataTable
        Try
            oclsDepartment = New clsDepartment()
            oclsDepartment.Cmp_id = Cmp_id
            dt = oclsDepartment.[Select_By_Cmp]()
            Return dt
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
#End Region
End Class
Public Class clsDepartment
    Inherits BaseClass
    Dim oClsDal As ClsDataccess = New ClsDataccess()
#Region "Property"

    Private _Department_id As Integer
    Private _Name As String
    Private _Value As Integer
    Private _Is_Active As String
    Private _Cmp_id As Integer
    Private _Ip_Address As String
    Private _Login_id As Integer
    Private _Department_category_id As Integer
    Private oclsDepartment As clsDepartment

    Public Property Login_id() As Integer
        Get
            Return _Login_id
        End Get
        Set(ByVal value As Integer)
            _Login_id = value
        End Set
    End Property

    Public Property Ip_Address() As String
        Get
            Return _Ip_Address
        End Get
        Set(ByVal value As String)
            _Ip_Address = value
        End Set
    End Property

    Public Property Cmp_id() As Integer
        Get
            Return _Cmp_id
        End Get
        Set(ByVal value As Integer)
            _Cmp_id = value
        End Set
    End Property

    Public Property Is_Active() As String
        Get
            Return _Is_Active
        End Get
        Set(ByVal value As String)
            _Is_Active = value
        End Set
    End Property

    Public Property Value() As Integer
        Get
            Return _Value
        End Get
        Set(ByVal value As Integer)
            _Value = value
        End Set
    End Property

    Public Property Name() As String
        Get
            Return _Name
        End Get
        Set(ByVal value As String)
            _Name = value
        End Set
    End Property

    Public Property Department_id() As Integer
        Get
            Return _Department_id
        End Get
        Set(ByVal value As Integer)
            _Department_id = value
        End Set
    End Property
    Public Property Department_category_id() As Integer
        Get
            Return _Department_category_id
        End Get
        Set(ByVal value As Integer)
            _Department_category_id = value
        End Set
    End Property
#End Region

    Public Function Insert() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@Department_id", Department_id, SqlDbType.Int)
            oColSqlparram.Add("@Name", Name)
            oColSqlparram.Add("@Value", Value, SqlDbType.Int)
            oColSqlparram.Add("@Cmp_id", Cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@Login_id", Login_id, SqlDbType.Int)
            oColSqlparram.Add("@Ip_Address", Ip_Address)
            oColSqlparram.Add("@Tran_Type", "I")
            oColSqlparram.Add("@department_category_id", Department_category_id, SqlDbType.Int)
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Department", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Update() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@Department_id", Department_id, SqlDbType.Int)
            oColSqlparram.Add("@Name", Name)
            oColSqlparram.Add("@Value", Value, SqlDbType.Int)
            oColSqlparram.Add("@Cmp_id", Cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@Login_id", Login_id, SqlDbType.Int)
            oColSqlparram.Add("@Ip_Address", Ip_Address)
            oColSqlparram.Add("@Tran_Type", "U")
            oColSqlparram.Add("@department_category_id", Department_category_id, SqlDbType.Int)
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Department", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function Delete() As Boolean
        Try
            Dim oColSqlparram As ColSqlparram = New ColSqlparram()
            oColSqlparram.Add("@Department_id", Department_id, SqlDbType.Int)
            oColSqlparram.Add("@Name", Name)
            oColSqlparram.Add("@Value", Value, SqlDbType.Int)
            oColSqlparram.Add("@Cmp_id", Cmp_id, SqlDbType.Int)
            oColSqlparram.Add("@Login_id", Login_id, SqlDbType.Int)
            oColSqlparram.Add("@Ip_Address", Ip_Address)
            oColSqlparram.Add("@Tran_Type", "D")
            oColSqlparram.Add("@department_category_id", Department_category_id, SqlDbType.Int)
            Dim dtlogin As DataTable = oClsDal.GetdataTableSp("P_M_Department", oColSqlparram)
            Return True
        Catch ex As Exception
            Throw New Exception(ex.Message)
        End Try
    End Function
    Public Function [Select]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@Department_id", Department_id, SqlDbType.Int)
        oColSqlparram.Add("@cmp_id", Cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_M_Department", oColSqlparram)

        Return dtlogin
    End Function
    Public Function [QrList]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()

        oColSqlparram.Add("@cmp_id", Cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("GetMachineList", oColSqlparram)

        Return dtlogin
    End Function
    Public Function [Select_By_Cmp]() As DataTable
        Dim ds As New DataSet

        Dim oColSqlparram As ColSqlparram = New ColSqlparram()
        oColSqlparram.Add("@cmp_id", Cmp_id, SqlDbType.Int)
        Dim dtlogin As DataTable = oClsDal.GetdataTableSp("Get_Department", oColSqlparram)

        Return dtlogin
    End Function
End Class

File: Converted_POS_08May2025/Converted_POS/Models/clsDepartment.cs
using Microsoft.Extensions.Configuration;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Threading.Tasks;

namespace Converted_POS.Models
{
    public class clsDepartment
    {
        public int? Department_id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public int? Value { get; set; }
        public bool Is_Active { get; set; }
        public int? Cmp_id { get; set; }
        public string Ip_Address { get; set; }
        public int? Login_id { get; set; }
        public int? Department_category_id { get; set; }
        public string department_category_name { get; set; }
        public DateTime? created_date { get; set; }
        public DateTime? modify_date { get; set; }
        public string Tran_Type { get; set; }
        public int? cmp_id { get; set; }
        public bool IsDepartmentInserted { get; set; } = true;
    }

    //public class clsDepartment_DAL
    //{
        //public static IConfiguration _configuration;
        //string connectionString = _configuration.GetConnectionString("POS_Connection");

        //public List<clsDepartment> bindDptCat(int? c_id, string conn)
        //{
        //    List<clsDepartment> dept = new List<clsDepartment>();

        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("Get_M_Deparment_Category", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", c_id);

        //        con.Open();
        //        SqlDataReader rdr = cmd.ExecuteReader();

        //        while (rdr.Read())
        //        {
        //            dept.Add(new clsDepartment
        //            {
        //                department_category_name = rdr["department_category_name"].ToString(),
        //                Department_category_id = Convert.ToInt32(rdr["department_category_id"])
        //            });
        //        }
        //    }

        //    return dept;
        //}



        //To View all dept details    
        //public IEnumerable<clsDepartment> SelectAll(int? c_id, int? d_id, string conn)
        //{
        //    List<clsDepartment> list = new List<clsDepartment>();

        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("Get_M_Department", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", c_id);
        //        cmd.Parameters.AddWithValue("@Department_id", d_id);

        //        con.Open();
        //        SqlDataReader rdr = cmd.ExecuteReader();

        //        while (rdr.Read())
        //        {
        //            clsDepartment dept = new clsDepartment();

        //            dept.Department_id = Convert.ToInt32(rdr["Department_id"]);
        //            dept.Name = rdr["Name"].ToString();
        //            dept.Value = Convert.ToInt32(rdr["value"].ToString());
        //            dept.Department_category_id = Convert.ToInt32(rdr["department_category_id"]);
        //            dept.department_category_name = rdr["department_category_name"].ToString();

        //            list.Add(dept);
        //        }
        //        con.Close();
        //    }
        //    return list;
        //}


        //To View all dept details    
        //public clsDepartment Select(int? c_id, int? d_id, string conn)
        //{
        //    clsDepartment list = new clsDepartment();

        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("Get_M_Department", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", c_id);
        //        cmd.Parameters.AddWithValue("@Department_id", d_id);

        //        con.Open();
        //        SqlDataReader rdr = cmd.ExecuteReader();

        //        while (rdr.Read())
        //        {
        //            clsDepartment dept = new clsDepartment();

        //            dept.Department_id = Convert.ToInt32(rdr["Department_id"]);
        //            dept.Name = rdr["Name"].ToString();
        //            dept.Value = Convert.ToInt32(rdr["value"].ToString());
        //            dept.Department_category_id = Convert.ToInt32(rdr["department_category_id"]);
        //            dept.department_category_name = rdr["department_category_name"].ToString();
                    
        //        }
        //        con.Close();
        //    }
        //    return list;
        //}


        //To Add new record    
        //public void Insert(clsDepartment cat, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Department", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
        //        cmd.Parameters.AddWithValue("@Department_id ", cat.Department_id);
        //        cmd.Parameters.AddWithValue("@Name", cat.Name);
        //        cmd.Parameters.AddWithValue("@Value", cat.Value);
        //        cmd.Parameters.AddWithValue("@Login_id ", 0);
        //        cmd.Parameters.AddWithValue("@Ip_Address", "");
        //        cmd.Parameters.AddWithValue("@department_category_id", cat.Department_category_id);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "I");

        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}

        //To Update record    
        //public void Update(clsDepartment cat, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Department", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
        //        cmd.Parameters.AddWithValue("@Department_id ", cat.Department_id);
        //        cmd.Parameters.AddWithValue("@Name", cat.Name);
        //        cmd.Parameters.AddWithValue("@Value", cat.Value);
        //        cmd.Parameters.AddWithValue("@Login_id ", cat.Login_id);
        //        cmd.Parameters.AddWithValue("@Ip_Address", cat.Ip_Address);
        //        cmd.Parameters.AddWithValue("@department_category_id", cat.Department_category_id);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "U");


        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}

        //To Delete record    
        //public void Delete(clsDepartment cat, string conn)
        //{
        //    using (SqlConnection con = new SqlConnection(conn))
        //    {
        //        SqlCommand cmd = new SqlCommand("P_M_Department", con);
        //        cmd.CommandType = CommandType.StoredProcedure;

        //        cmd.Parameters.AddWithValue("@cmp_id", cat.cmp_id);
        //        cmd.Parameters.AddWithValue("@Department_id ", cat.Department_id);
        //        cmd.Parameters.AddWithValue("@Name", cat.Name);
        //        cmd.Parameters.AddWithValue("@Value", cat.Value);
        //        cmd.Parameters.AddWithValue("@Login_id ", cat.Login_id);
        //        cmd.Parameters.AddWithValue("@Ip_Address", cat.Ip_Address);
        //        cmd.Parameters.AddWithValue("@department_category_id", cat.Department_category_id);
        //        cmd.Parameters.AddWithValue("@Tran_Type", "D");


        //        con.Open();
        //        cmd.ExecuteNonQuery();
        //        con.Close();
        //    }
        //}


    //}
}
